compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/TranslatingVersionNessieApi.java

```diff
@@ -408,11 +412,24 @@ private <T> T createProxy(
             Thread.currentThread().getContextClassLoader(),
             interfaces,
             (proxyInstance, method, args) -> {
-              Method targetMethod =
-                  o.getClass()
-                      .getMethod(
-                          method.getName(),
-                          translateTypes(reverseClassLoader, method.getParameterTypes()));
+              Method targetMethod;
+              try {
+                targetMethod =
+                    o.getClass()
+                        .getMethod(
+                            method.getName(),
+                            translateTypes(reverseClassLoader, method.getParameterTypes()));
+              } catch (NoSuchMethodException e) {
+                throw new RuntimeException(
```

Comment:
This change gives a better test error message, when trying to invoke a client-API method, that doesn't exist in older client API versions.